components/engine_traits/src/engine.rs

```diff
@@ -62,6 +65,86 @@ pub trait KvEngine:
 ///
 // It should be named as `EngineFactory` for consistency, but we are about to rename
 // engine to tablet, so always use tablet for new traits/types.
-pub trait TabletFactory<EK> {
-    fn create_tablet(&self) -> Result<EK>;
+pub trait TabletFactory<EK>: Clone {
```

Comment:
This is not a good idea. It makes `Factory` not object safe. `Factory` is not in performance critical path, making it object safe can bring a lot of convenience.

Reply:
@BusyJay agree. I realized it won't work for dyn TabletFactory